main.c

```diff
@@ -247,6 +247,7 @@ add_module_solvables (Repo           *repo,
                              0);
 
       add_module_dependencies (pool, solvable, deps);
+      pool_createwhatprovides (pool);
```

Comment:
why is it needed here?

Reply:
Without this libsolv was crashing because `pool->whatprovides_rel` was NULL.

Reply:
right, add_module_rpm_artifacts rely on provides being created.. ideally we would add just new provides, but we can deal with it later